# Assignment 6

*(Due Saturday, October 26, 2019 at noon.)*

**1)** Below, paste the code needed to import and clean your file from the `../dataset` directory in your repository, then save the result to a new file. I should be able to run the code in this notebook and get my own copy of your cleaned, processed data.

**Note:** Your assignment file will be publicly accessible on Github. If your dataset is private, don't print anything confidential in this assignment. 

In [21]:
import numpy as np
import pandas as pd

In [22]:
#loading my environmental data
###set the first line as col names
banner4=pd.read_csv('../AStahl Dataset/PhysicalData_9.23.2019 .csv', nrows=5, encoding='latin')
colnames4 = list(banner4.columns)
print(colnames4)
###read in dataset with the col names that was just made
environ_data=pd.read_csv('../AStahl Dataset/PhysicalData_9.23.2019 .csv', encoding='latin', 
                         names=colnames4, skiprows=2)
###remove the last unwated 'notes' rows/unwanted data
environ_data=environ_data.iloc[:,:-7]
###only pulling the columns I want to use 
environ_data=environ_data.iloc[:,[3,6,7]]
environ_data.head()
####GOOD TO GO

['ID', 'Sample ID', 'Sample location', 'Date', 'Time', 'Weather desc', 'Surface Temp', 'Bottom Temp', 'Air Temp', 'Hours before high tide', 'Surface Salinity', 'Bottom Salinity', 'Secchi Depth', 'Unnamed: 13', 'Unnamed: 14', 'Unnamed: 15']


Date  Surface Temp Bottom Temp
0  25-Jan-99          0.00           0
1  01-Feb-99          2.25         2.5
2  08-Feb-99          0.00           0
3  15-Feb-99          0.00           0
4  22-Feb-99          0.00           0

In [23]:
#loading my chlorophyll data 
###set the first line as col names
banner5=pd.read_csv('../AStahl Dataset/chldata_9.23.2019.csv')
colnames5 = list(banner5.columns)
print(colnames5)
###read in dataset with the col names that was just made
chl_data=pd.read_csv('../AStahl Dataset/chldata_9.23.2019.csv', 
                     names=colnames5,skiprows=3)
###remove the last unwated 'notes' rows
chl_data=chl_data.iloc[:,:-17]
###only pulling the columns I want to use
chl_data=chl_data.iloc[:,[12,13,14,15,16]]
##Rename columns
chl_data.rename(columns={'Sample Date.1':'Date','surface chla all':'Surface Chla all',
                        'Surface Phaeo all.1':'Surface Phaeo all','Bottom Chla all.1':'Bottom Chla all',
                         'Bottom Phaeo all.1':'Bottom Phaeo all'}, inplace=True)
chl_data.head()
###GOOD TO GO

['Sample', 'Sample location', 'Sample Date', 'Surface Chla all', 'Surface Phaeo all', 'Bottom Chla all', 'Bottom Phaeo all', 'surface chla <20 um', 'Surface Phaeo <20 um', 'Bottom Chla <20 um', 'Bottom Phaeo <20 um', 'Unnamed: 11', 'Sample Date.1', 'surface chla all', 'Surface Phaeo all.1', 'Bottom Chla all.1', 'Bottom Phaeo all.1', 'surface chla <20 um.1', 'Surface Phaeo <20 um.1', 'Bottom Chla <20 um.1', 'Bottom Phaeo <20 um.1', 'Unnamed: 21', 'Unnamed: 22', 'Unnamed: 23', 'Unnamed: 24', 'Unnamed: 25', 'Unnamed: 26', 'Unnamed: 27', 'Unnamed: 28', 'Unnamed: 29', 'Unnamed: 30', 'Unnamed: 31', 'Unnamed: 32', 'Unnamed: 33', 'Unnamed: 34', 'Unnamed: 35', 'Unnamed: 36', 'Unnamed: 37']


Date Surface Chla all Surface Phaeo all Bottom Chla all  \
0  25-Jan-99             3.72              0.80            3.26   
1  01-Feb-99             2.90              0.74            2.66   
2  08-Feb-99             2.20              0.94            3.24   
3  15-Feb-99             1.34              1.20            1.47   
4  22-Feb-99            12.56              5.33            9.04   

  Bottom Phaeo all  
0             1.15  
1             0.83  
2             1.98  
3             0.60  
4             4.97

In [24]:
#loading my silica data
###set the first line as col names
banner6=pd.read_excel('../AStahl Dataset/nutrientdata.xls',sheet_name=1)
colnames6=list(banner6.columns)
print(colnames6)
###read in dataset with the col names that was just made
Si_data=pd.read_excel('../AStahl Dataset/nutrientdata.xls',sheet_name=1, names=colnames6,
                     skiprows=2)
#rename columns
Si_data.rename(columns={'DATE':'Date','Si':'Surface Si','Si.1':'Bottom Si'}, inplace=True)
##remove unwanted columns by keeping the columns you want 
Si_data=Si_data.iloc[:,[1,3,9]]
Si_data.head()
###GOOD TO GO

['Sample ID', 'DATE', 'Surface', 'Si', 'Unnamed: 4', 'Unnamed: 5', 'Sample ID.1', 'DATE.1', 'Deep', 'Si.1']


Date Surface Si Bottom Si
0 2003-02-24   0.134507   0.11277
1 2003-03-10   0.779352  0.746106
2 2003-03-24    3.54383   2.39175
3 2003-04-09     2.6703  0.863474
4 2003-04-23    1.20083  0.579066

In [25]:
#loading my nitrogen and phosphorus data
###set the fifth line as col names
banner7=pd.read_excel('../AStahl Dataset/nutrientdata.xls',sheet_name=0, header=4)
colnames7=list(banner7.columns)
print(colnames7)
###read in dataset with the col names that was just made
NandP_data=pd.read_excel('../AStahl Dataset/nutrientdata.xls',sheet_name=0, names=colnames7,
                         skiprows=6)
###rename columns
NandP_data.rename(columns={'Unnamed: 0':'Date','NH4':'Surface NH4','DIP':'Surface DIP',
                           'NO3+2':'Surface NO3+2','NO3':'Surface NO3','NO2':'Surface NO2',
                          'DIN':'Surface DIN','NH4.1':'Bottom NH4','DIP.1':'Bottom DIP',
                           'NO3+2.1':'Bottom NO3+2','NO3.1':'Bottom NO3','NO2.1':'Bottom NO2',
                           'DIN.1':'Bottom DIN'}, inplace=True)
##remove unwanted columns by keeping the columns you want 
NandP_data=NandP_data.iloc[:,[0,2,3,4,5,6,7,11,12,13,14,15,16]]
NandP_data.head()
##GOOD TO GO

['Unnamed: 0', 'Unnamed: 1', 'NH4', 'DIP', 'NO3+2', 'NO3', 'NO2', 'DIN', 'Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10', 'NH4.1', 'DIP.1', 'NO3+2.1', 'NO3.1', 'NO2.1', 'DIN.1']


Date Surface NH4 Surface DIP Surface NO3+2 Surface NO3 Surface NO2  \
0 2003-05-05     1.17807    0.171054       2.31024     2.09136    0.218876   
1 2003-05-21    0.241243   0.0941145      0.108927   0.0469255   0.0620015   
2 2003-05-28     3.85202    0.440969       2.26947     2.07585    0.193621   
3 2003-06-04    0.546374    0.261893       1.38173     1.12293    0.258808   
4 2003-06-09    0.752151    0.240058       2.29541      1.9335    0.361901   

   Surface DIN Bottom NH4 Bottom DIP Bottom NO3+2 Bottom NO3 Bottom NO2  \
0     3.488308   0.679138   0.269104     0.329064   0.280104    0.04896   
1     0.350170   0.510182   0.241036     0.146274   0.105181   0.041093   
2     6.121488    3.94511   0.560924      1.36987    1.21594   0.153929   
3     1.928109    4.20001   0.691386      2.22854    1.95345   0.275088   
4     3.047557    2.71059   0.542193     0.943768   0.780786   0.162982   

  Bottom DIN  
0     1.0082  
1   0.656456  
2    5.31499  
3    6.42855  
4    3.65435

In [26]:
#loading my count data
###set the first line as col names
banner8=pd.read_excel('../AStahl Dataset/countdata_9.23.2019 .xls',sheet_name=1)
colnames8=list(banner8.columns)
###read in dataset with the col names that was just made
count_data=pd.read_excel('../AStahl Dataset/countdata_9.23.2019 .xls',sheet_name=1, 
                         names=colnames8, skiprows=1)
#rename columns
count_data.rename(columns={'DATE':'Date'}, inplace=True)
##remove unwanted columns by keeping the columns you want 
count_data=count_data.iloc[:,[0,3,8,214,215,216,217,218,234,245,248]]
count_data.head()

Date  Total abundance  Akashiwo sanguineum  \
0  1999-01-25 00:00:00         124000.0                  0.0   
1  1999-02-01 00:00:00         113000.0                  0.0   
2  1999-02-08 00:00:00         107000.0                  0.0   
3  1999-02-15 00:00:00          95500.0                  0.0   
4  1999-02-22 00:00:00         163500.0                  0.0   

  Pseudo-nitzschia fraudulenta  Pseudonitzschia pseudodelicatissima  \
0                            0                                  0.0   
1                            0                                  0.0   
2                            0                                  0.0   
3                            0                                  0.0   
4                            0                                  0.0   

   Pseudonitzschia pungens  Pseudo-nitzschia spp.  \
0                      0.0                    0.0   
1                      0.0                    0.0   
2                      0.0                    0.0   
3                      0.0                    0.0   
4                      0.0                    0.0   

   Pseudo-nitzschia turgidula cf  Skeletonema spp.  \
0                            0.0           14000.0   
1                            0.0           12000.0   
2                            0.0            3000.0   
3                            0.0               0.0   
4                            0.0            6000.0   

   Thalassiosira nordenskioeldii  Thalassiosira rotula  
0                            0.0                   0.0  
1                            0.0                   0.0  
2                            0.0                   0.0  
3                            0.0                   0.0  
4                            0.0                   0.0

In [36]:
#merging some above datasets
data1=pd.merge(environ_data, chl_data, on=['Date'])
data1.head()
#fixing the Date format for environ and chl
data1['Date'] =  pd.to_datetime(data1['Date'],
                              format='%d-%b-%y')
data1
print(data1['Date'])

0     1999-01-25
1     1999-02-01
2     1999-02-08
3     1999-02-15
4     1999-02-22
5     1999-03-01
6     1999-03-10
7     1999-03-16
8     1999-03-24
9     1999-03-29
10    1999-04-05
11    1999-04-12
12    1999-04-19
13    1999-04-26
14    1999-05-03
15    1999-05-10
16    1999-05-17
17    1999-05-26
18    1999-05-31
19    1999-06-07
20    1999-06-14
21    1999-06-21
22    1999-06-28
23    1999-07-05
24    1999-07-12
25    1999-07-19
26    1999-08-09
27    1999-08-19
28    1999-08-23
29    1999-09-02
         ...    
894   2018-12-17
895   2018-12-26
896   2018-12-31
897   2019-01-14
898   2019-01-22
899   2019-01-28
900   2019-02-11
901   2019-02-18
902   2019-02-27
903   2019-03-11
904   2019-03-18
905   2019-03-25
906   2019-04-17
907   2019-04-22
908   2019-04-29
909   2019-05-13
910   2019-05-20
911   2019-05-28
912   2019-06-10
913   2019-06-17
914   2019-06-24
915   2019-07-15
916   2019-07-22
917   2019-07-29
918   2019-08-05
919   2019-08-12
920   2019-08-20
921   2019-08-

In [37]:
#merging more datasets
data2=pd.merge(Si_data, NandP_data, on=['Date'])
data2.head(15)

#fixing the date format for Si and N/P
data2['Date'] =  pd.to_datetime(data2['Date'],
                              format='%Y-%m-%d')
print(data2['Date'])

0            NaT
1            NaT
2            NaT
3            NaT
4            NaT
5            NaT
6     2003-05-21
7     2003-05-28
8     2003-06-04
9     2003-06-09
10    2003-06-16
11    2003-06-23
12    2003-06-30
13    2003-07-09
14    2003-07-14
15    2003-07-21
16    2003-07-28
17    2003-08-04
18    2003-08-13
19    2003-08-20
20    2003-08-27
21    2003-09-03
22    2003-09-10
23    2003-09-17
24    2003-09-24
25    2003-09-29
26    2003-10-08
27    2003-10-15
28    2003-10-22
29    2003-10-29
         ...    
531   2019-02-11
532   2019-02-18
533   2019-02-27
534   2019-03-06
535   2019-03-11
536   2019-03-18
537   2019-03-25
538   2019-04-01
539   2019-04-08
540   2019-04-17
541   2019-04-22
542   2019-04-29
543   2019-05-06
544   2019-05-13
545   2019-05-20
546   2019-05-28
547   2019-06-03
548   2019-06-10
549   2019-06-17
550   2019-06-24
551   2019-07-01
552   2019-07-08
553   2019-07-15
554   2019-07-22
555   2019-07-29
556   2019-08-05
557   2019-08-12
558   2019-08-

In [29]:
#merging the first 4 datasets
data3=pd.merge(data1,data2, on=['Date'])
data3

Date  Surface Temp Bottom Temp Surface Chla all Surface Phaeo all  \
0   2003-05-21         12.90        11.8             5.57              0.33   
1   2003-05-28         12.30        11.9             0.86              0.61   
2   2003-06-04         14.10          13             9.24              0.67   
3   2003-06-09         15.90        14.9            12.60              2.73   
4   2003-06-16         15.50         NaN             9.47              2.06   
5   2003-06-23         16.90        16.2             5.71             10.69   
6   2003-06-30         19.90        16.8            10.37              2.13   
7   2003-07-09         23.00        18.3             4.18              1.65   
8   2003-07-14         20.40        18.4             6.31              2.02   
9   2003-07-21         22.00        18.9             0.57              3.98   
10  2003-07-28         22.00          21             3.16              2.99   
11  2003-08-04         23.00        21.5             4.10              1.50   
12  2003-08-13         24.50        22.5             4.30              1.26   
13  2003-08-20         22.80        21.6             7.18              2.08   
14  2003-08-27         21.80        19.2            10.57              1.44   
15  2003-09-03         19.00          19             5.16              1.80   
16  2003-09-10         19.40        19.2            15.33              1.60   
17  2003-09-17         20.20          20             4.88              1.84   
18  2003-09-29         20.30        20.1             6.04              1.81   
19  2003-10-08         17.10        16.9             4.77              1.28   
20  2003-10-15         16.50        16.3             3.67              1.44   
21  2003-10-22         14.00        13.6             2.60              0.89   
22  2003-10-29         13.70        13.3             2.36              0.94   
23  2003-11-10         11.20        11.4             2.60              0.93   
24  2003-11-17          8.20         9.6             1.86              0.38   
25  2003-11-24          9.60         9.2             2.12              0.77   
26  2003-12-01          8.80         8.8             2.58              0.76   
27  2003-12-09          3.40         4.4             1.69              0.76   
28  2003-12-15          3.60           4             1.18              0.50   
29  2003-12-30          5.00         5.6             4.36              0.72   
..         ...           ...         ...              ...               ...   
603 2018-11-26          7.31        7.88             1.79              0.22   
604 2018-12-10          5.09         7.4             2.74              0.56   
605 2018-12-17          5.56         6.2             3.56              1.05   
606 2018-12-26          4.90           5             3.39             -0.94   
607 2018-12-31          5.78        6.43             5.69              0.56   
608 2019-01-14          2.84        3.21             5.35              0.12   
609 2019-01-22          2.62        2.63            16.22              2.82   
610 2019-01-28          1.48        3.46            28.34             -4.36   
611 2019-02-11          1.90        2.32            18.19             -1.35   
612 2019-02-18          2.92        3.62             2.58              0.78   
613 2019-02-27          2.79        3.15             3.59              0.54   
614 2019-03-11          2.54        2.85             8.59              0.44   
615 2019-03-18          4.23        4.74             0.80              0.39   
616 2019-03-25          5.22        5.17             2.40              1.04   
617 2019-04-17          7.90         8.5             2.31              0.67   
618 2019-04-22         10.75        9.87             1.94             -0.07   
619 2019-04-29         10.39        8.89             3.24              1.04   
620 2019-05-13           NaN         NaN             3.96              0.02   
621 2019-05-20         12.54       12.34             2.63   

In [30]:
#fixing the count data date format
count_data['Date']= pd.to_datetime(count_data['Date'],errors='coerce')
count_data.info()
count_data.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1585 entries, 0 to 1584
Data columns (total 11 columns):
Date                                   1578 non-null datetime64[ns]
Total abundance                        1576 non-null float64
Akashiwo sanguineum                    1580 non-null float64
Pseudo-nitzschia fraudulenta           1580 non-null object
Pseudonitzschia pseudodelicatissima    1580 non-null float64
Pseudonitzschia pungens                1580 non-null float64
Pseudo-nitzschia spp.                  1573 non-null float64
Pseudo-nitzschia turgidula cf          1580 non-null float64
Skeletonema spp.                       1580 non-null float64
Thalassiosira nordenskioeldii          1580 non-null float64
Thalassiosira rotula                   1570 non-null float64
dtypes: datetime64[ns](1), float64(9), object(1)
memory usage: 136.3+ KB


Date  Total abundance  Akashiwo sanguineum  \
0 1999-01-25         124000.0                  0.0   
1 1999-02-01         113000.0                  0.0   
2 1999-02-08         107000.0                  0.0   
3 1999-02-15          95500.0                  0.0   
4 1999-02-22         163500.0                  0.0   

  Pseudo-nitzschia fraudulenta  Pseudonitzschia pseudodelicatissima  \
0                            0                                  0.0   
1                            0                                  0.0   
2                            0                                  0.0   
3                            0                                  0.0   
4                            0                                  0.0   

   Pseudonitzschia pungens  Pseudo-nitzschia spp.  \
0                      0.0                    0.0   
1                      0.0                    0.0   
2                      0.0                    0.0   
3                      0.0                    0.0   
4                      0.0                    0.0   

   Pseudo-nitzschia turgidula cf  Skeletonema spp.  \
0                            0.0           14000.0   
1                            0.0           12000.0   
2                            0.0            3000.0   
3                            0.0               0.0   
4                            0.0            6000.0   

   Thalassiosira nordenskioeldii  Thalassiosira rotula  
0                            0.0                   0.0  
1                            0.0                   0.0  
2                            0.0                   0.0  
3                            0.0                   0.0  
4                            0.0                   0.0

In [31]:
#final merge of all datasets
data=pd.merge(data3,count_data, on=['Date'])
data.head()

Date  Surface Temp Bottom Temp Surface Chla all Surface Phaeo all  \
0 2003-05-21          12.9        11.8             5.57              0.33   
1 2003-05-28          12.3        11.9             0.86              0.61   
2 2003-06-04          14.1          13             9.24              0.67   
3 2003-06-09          15.9        14.9            12.60              2.73   
4 2003-06-16          15.5         NaN             9.47              2.06   

  Bottom Chla all Bottom Phaeo all Surface Si Bottom Si Surface NH4  ...  \
0            4.79             0.83   0.269958   2.33956    0.241243  ...   
1            1.19             1.35    9.06109   7.63842     3.85202  ...   
2            4.38             1.34    27.7229   17.7328    0.546374  ...   
3            5.28             0.92    23.7053   22.2505    0.752151  ...   
4             NaN              NaN     26.176       NaN     1.07933  ...   

  Total abundance Akashiwo sanguineum Pseudo-nitzschia fraudulenta  \
0         52000.0                 0.0                            0   
1         15000.0                 0.0                            0   
2         47000.0                 0.0                            0   
3         41000.0                 0.0                            0   
4         44000.0                 0.0                            0   

  Pseudonitzschia pseudodelicatissima  Pseudonitzschia pungens  \
0                                 0.0                      0.0   
1                                 0.0                      0.0   
2                                 0.0                      0.0   
3                                 0.0                      0.0   
4                                 0.0                      0.0   

  Pseudo-nitzschia spp. Pseudo-nitzschia turgidula cf Skeletonema spp.  \
0                   0.0                           0.0          15000.0   
1                   0.0                           0.0           5000.0   
2                   0.0                           0.0           9000.0   
3                   0.0                           0.0              0.0   
4                   0.0                           0.0              0.0   

  Thalassiosira nordenskioeldii Thalassiosira rotula  
0                           0.0                  0.0  
1                           0.0                  0.0  
2                           0.0                  0.0  
3                           0.0                  0.0  
4                           0.0                  0.0  

[5 rows x 31 columns]

In [32]:
#I am not looking to fill in any data so I don't want to use the  fillna function
#If I am looking to compare changes over time, I need all the values for all of the variables
#therefore I think the best way to be able to collect all that data is to remove the entire row
data.dropna(how='all')
# I do not think that the data.dropna() will give me enough resolution for the test I want to do 
#Can I do statistical analysis with the data if there is NaN in it? Or does NaN HAVE to be removed 
     ##to do statistical analysis?

Date  Surface Temp Bottom Temp Surface Chla all Surface Phaeo all  \
0    2003-05-21         12.90        11.8             5.57              0.33   
1    2003-05-28         12.30        11.9             0.86              0.61   
2    2003-06-04         14.10          13             9.24              0.67   
3    2003-06-09         15.90        14.9            12.60              2.73   
4    2003-06-16         15.50         NaN             9.47              2.06   
5    2003-06-23         16.90        16.2             5.71             10.69   
6    2003-06-30         19.90        16.8            10.37              2.13   
7    2003-07-09         23.00        18.3             4.18              1.65   
8    2003-07-14         20.40        18.4             6.31              2.02   
9    2003-07-21         22.00        18.9             0.57              3.98   
10   2003-07-28         22.00          21             3.16              2.99   
11   2003-08-04         23.00        21.5             4.10              1.50   
12   2003-08-13         24.50        22.5             4.30              1.26   
13   2003-08-20         22.80        21.6             7.18              2.08   
14   2003-08-27         21.80        19.2            10.57              1.44   
15   2003-09-03         19.00          19             5.16              1.80   
16   2003-09-10         19.40        19.2            15.33              1.60   
17   2003-09-17         20.20          20             4.88              1.84   
18   2003-09-29         20.30        20.1             6.04              1.81   
19   2003-10-08         17.10        16.9             4.77              1.28   
20   2003-10-15         16.50        16.3             3.67              1.44   
21   2003-10-22         14.00        13.6             2.60              0.89   
22   2003-10-29         13.70        13.3             2.36              0.94   
23   2003-11-10         11.20        11.4             2.60              0.93   
24   2003-11-17          8.20         9.6             1.86              0.38   
25   2003-11-24          9.60         9.2             2.12              0.77   
26   2003-12-01          8.80         8.8             2.58              0.76   
27   2003-12-09          3.40         4.4             1.69              0.76   
28   2003-12-15          3.60           4             1.18              0.50   
29   2003-12-30          5.00         5.6             4.36              0.72   
...         ...           ...         ...              ...               ...   
1835 2019-04-17          7.90         8.5             2.31              0.67   
1836 2019-04-22         10.75        9.87             1.94             -0.07   
1837 2019-04-22         10.75        9.87             1.94             -0.07   
1838 2019-04-29         10.39        8.89             3.24              1.04   
1839 2019-04-29         10.39        8.89             3.24              1.04   
1840 2019-05-13           NaN         NaN             3.96              0.02   
1841 2019-05-13           NaN         NaN             3.96              0.02   
1842 2019-05-20         12.54       12.34             2.63              1.38   
1843 2019-05-20         12.54       12.34             2.63              1.38   
1844 2019-05-28         16.73       16.67             4.45             -0.27   
1845 2019-05-28         16.73       16.67             4.45             -0.27   
1846 2019-06-10         17.50       15.57             2.29              1.96   
1847 2019-06-10         17.50       15.57             2.29              1.96   
1848 2019-06-17         17.78       17.48             4.44              2.19   
1849 2019-06-17         17.78       17.48             4.44              2.19   
1850 2019-06-24         19.31        17.5             8.75              2.99   
1851 2019-06-24         19.31        17.5             8.75              2.99   
1852 2019-07-15         23.47       21.43             5.53              2.28   
1853 2019-0

In [33]:
#removing duplicates
data.duplicated()
data.drop_duplicates()

Date  Surface Temp Bottom Temp Surface Chla all Surface Phaeo all  \
0    2003-05-21         12.90        11.8             5.57              0.33   
1    2003-05-28         12.30        11.9             0.86              0.61   
2    2003-06-04         14.10          13             9.24              0.67   
3    2003-06-09         15.90        14.9            12.60              2.73   
4    2003-06-16         15.50         NaN             9.47              2.06   
5    2003-06-23         16.90        16.2             5.71             10.69   
6    2003-06-30         19.90        16.8            10.37              2.13   
7    2003-07-09         23.00        18.3             4.18              1.65   
8    2003-07-14         20.40        18.4             6.31              2.02   
9    2003-07-21         22.00        18.9             0.57              3.98   
10   2003-07-28         22.00          21             3.16              2.99   
11   2003-08-04         23.00        21.5             4.10              1.50   
12   2003-08-13         24.50        22.5             4.30              1.26   
13   2003-08-20         22.80        21.6             7.18              2.08   
14   2003-08-27         21.80        19.2            10.57              1.44   
15   2003-09-03         19.00          19             5.16              1.80   
16   2003-09-10         19.40        19.2            15.33              1.60   
17   2003-09-17         20.20          20             4.88              1.84   
18   2003-09-29         20.30        20.1             6.04              1.81   
19   2003-10-08         17.10        16.9             4.77              1.28   
20   2003-10-15         16.50        16.3             3.67              1.44   
21   2003-10-22         14.00        13.6             2.60              0.89   
22   2003-10-29         13.70        13.3             2.36              0.94   
23   2003-11-10         11.20        11.4             2.60              0.93   
24   2003-11-17          8.20         9.6             1.86              0.38   
25   2003-11-24          9.60         9.2             2.12              0.77   
26   2003-12-01          8.80         8.8             2.58              0.76   
27   2003-12-09          3.40         4.4             1.69              0.76   
28   2003-12-15          3.60           4             1.18              0.50   
29   2003-12-30          5.00         5.6             4.36              0.72   
...         ...           ...         ...              ...               ...   
1835 2019-04-17          7.90         8.5             2.31              0.67   
1836 2019-04-22         10.75        9.87             1.94             -0.07   
1837 2019-04-22         10.75        9.87             1.94             -0.07   
1838 2019-04-29         10.39        8.89             3.24              1.04   
1839 2019-04-29         10.39        8.89             3.24              1.04   
1840 2019-05-13           NaN         NaN             3.96              0.02   
1841 2019-05-13           NaN         NaN             3.96              0.02   
1842 2019-05-20         12.54       12.34             2.63              1.38   
1843 2019-05-20         12.54       12.34             2.63              1.38   
1844 2019-05-28         16.73       16.67             4.45             -0.27   
1845 2019-05-28         16.73       16.67             4.45             -0.27   
1846 2019-06-10         17.50       15.57             2.29              1.96   
1847 2019-06-10         17.50       15.57             2.29              1.96   
1848 2019-06-17         17.78       17.48             4.44              2.19   
1849 2019-06-17         17.78       17.48             4.44              2.19   
1850 2019-06-24         19.31        17.5             8.75              2.99   
1851 2019-06-24         19.31        17.5             8.75              2.99   
1852 2019-07-15         23.47       21.43             5.53              2.28   
1853 2019-0

In [34]:
#Save your file (change the name if necessary).
data.to_csv('../AStahl Dataset/all_data.csv')

**2)** A brief survey: Have you taken an introductory statistics course, and if so, do you remember much from it? For example,

* Can you articulate the null hypotheses for the questions you formulated last week involving your data? 
* Do you know what a t-test is for?
* Do you know what a *p* value represents?
* Do you know what standard deviation is, and how it's calculated?

I took an undergraduate 'Introduction to statistics' class as well as a graduate 'Ecological Statistics' course. 
The null hypothesis for my questions is that there will be no change in temperature/environmental factors or cell abundance over time. 
A t-test is used to determine if there is a statistical difference between the means of 2 groups. 
A p-value is the set level used to determine if there is a significant statistical result from your analysis (usually 0.05). The p-value is the main source of information for accepting or rejecting your null hypothesis. 
The standard deviation tells you how far your data spreads out around the mean. To calculate it:
     find the mean of your dataset
     subtract the mean from each data point 
     square the result
     find the mean of the squared results (this number is known as the variance)
     take the square root of the variance and that is your SD. 

#### Submitting

In the Terminal (Mac) or Git Bash (Windows):

`git commit -a -m 'Assignment 6 completed'`  
`git push`

### Reading

1. [_Python for Data Analysis_](https://uri-primo.hosted.exlibrisgroup.com/permalink/f/10nopmq/01URI_ALMA51209454630002396):
    * Section 7.3
    * Chapter 10